src/azul/__init__.py

```diff
@@ -1149,16 +1154,17 @@ def aggregation_lambda_timeout(self, *, retry: bool) -> int:
     def api_gateway_lambda_timeout(self) -> int:
         return self.api_gateway_timeout + self.api_gateway_timeout_padding
 
-    # This is set dynamically at runtime
+    # These attributes are  set dynamically at runtime
     lambda_context: Optional[LambdaContext] = None
+    in_rest_api: bool = False
 
     @property
     def _timing_is_restricted(self) -> bool:
-        if self.lambda_context is None:
-            return False
-        else:
+        if self.in_rest_api:
             remaining = self.lambda_context.get_remaining_time_in_millis() / 1000
             return remaining <= self.api_gateway_lambda_timeout
```

Comment:
The previous implementation was odd. In non-API Gateway lambdas like the indexer ones, it would be false up to 31s before that lambda would time out. With the added `if self.in_rest_api` guard in place it can never be false. I think we can remove `Config.lambda_context` and everything supporting it.

```diff
Subject: [PATCH] Fix: Diagram of shared deployments is inaccurate (#5391)
---
Index: src/azul/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/__init__.py b/src/azul/__init__.py
--- a/src/azul/__init__.py	(revision 2296f29270011de153ba0fef4b2015b84ebde747)
+++ b/src/azul/__init__.py	(date 1700008919406)
@@ -1154,17 +1154,12 @@
     def api_gateway_lambda_timeout(self) -> int:
         return self.api_gateway_timeout + self.api_gateway_timeout_padding
 
-    # These attributes are  set dynamically at runtime
-    lambda_context: Optional[LambdaContext] = None
-    in_rest_api: bool = False
+    # This attribute is set dynamically at runtime
+    lambda_is_handling_api_gateway_request: bool = False
 
     @property
     def _timing_is_restricted(self) -> bool:
-        if self.in_rest_api:
-            remaining = self.lambda_context.get_remaining_time_in_millis() / 1000
-            return remaining <= self.api_gateway_lambda_timeout
-        else:
-            return False
+        return self.lambda_is_handling_api_gateway_request
 
     @property
     def terra_client_timeout(self) -> float:
Index: src/azul/chalice.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/chalice.py b/src/azul/chalice.py
--- a/src/azul/chalice.py	(revision 2296f29270011de153ba0fef4b2015b84ebde747)
+++ b/src/azul/chalice.py	(date 1700008919409)
@@ -137,8 +137,7 @@
         # Middleware is invoked in order of registration
         self.register_middleware(self._logging_middleware, 'http')
         self.register_middleware(self._hsts_header_middleware, 'http')
-        self.register_middleware(self._lambda_context_middleware, 'all')
-        self.register_middleware(self._rest_api_context_middleware, 'http')
+        self.register_middleware(self._api_gateway_context_middleware, 'http')
         self.register_middleware(self._authentication_middleware, 'http')
 
     def __call__(self, event: dict, context: LambdaContext) -> dict[str, Any]:
@@ -186,19 +185,12 @@
             response = get_response(event)
         return response
 
-    def _lambda_context_middleware(self, event, get_response):
-        config.lambda_context = self.lambda_context
+    def _api_gateway_context_middleware(self, event, get_response):
+        config.lambda_is_handling_api_gateway_request = True
         try:
             return get_response(event)
         finally:
-            config.lambda_context = None
-
-    def _rest_api_context_middleware(self, event, get_response):
-        config.in_rest_api = True
-        try:
-            return get_response(event)
-        finally:
-            config.in_rest_api = False
+            config.lambda_is_handling_api_gateway_request = False
 
     def _hsts_header_middleware(self, event, get_response):
         """
```